{{cookiecutter.project_name}}/monitoring/SetUpDataMonitoringForInferenceTable.py

```diff
@@ -0,0 +1,118 @@
+granularities = ["1 hour"]
+
+# List of Databricks entity names that are associated
+# with this table. Only following entities are supported:
+#    ["models:/registry_model_name"] links a model registry model to the monitored table.
+linked_entities = ["models:/prod-{{cookiecutter.model_name}}"]
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC 
+# MAGIC ### Optional Fields
+# MAGIC TODO: Fill in optional fields as needed
+
+# COMMAND ----------
+
+# The name of the column that holds the label for this row. This field is optional if you do not have labels to provide with your data.
+# This column will be used to compute model quality metrics
+label_col = None
+
+# The name of the column that holds the example ID for this row. This might be a unique identifier attached to each inference 
+# or some other identifier you don’t want analyzed as a feature. The provided column will be ignored during analysis. 
+example_id_col = None
+
+# A table containing baseline data for comparison. The baseline table is expected to match the schema of the monitored table
+# If columns are missing on either side then monitoring will use best-effort heuristics to compute the output metrics.
+baseline_table_name = None
+
+# data_monitoring_dir = "/Users/first.last@company.com" The absolute path to user-configurable directory for storing data monitoring assets.
+# The path must be pointing to an existing directory. If provided, the assets will be created under:
+#         "{data_monitoring_dir}/{table_name}"
+# Otherwise, the assets will be stored under the default directory:
+#         "/Users/{user_name}/databricks_data_monitoring/{table_name}"
+# Note that this directory can exist anywhere, including "/Shared/" or other locations outside of the /Users/ 
+# directory; this can be useful when configuring a production monitor intended to be shared within an organization.
+data_monitoring_dir = None
+
+# List of column expressions to slice data with for targeted analysis. The data is grouped by each expression independently,
+# resulting in a separate slice for each predicate and its complements. For example `slicing_exprs=["col_1", "col_2 > 10"]` will generate the
+# following slices: two slices for `col_2 > 10` (True and False), and one slice per unique value in `col1`. For high-cardinality columns, only the
+# top 100 unique values by frequency will generate slices.
+slicing_exprs = None
+
+# A list of custom metrics to compute alongside existing aggregate, derived, and drift metrics.
+custom_metrics = []
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC 
+# MAGIC ### Create a monitor
+
+# COMMAND ----------
+
+import databricks.data_monitoring as dm
+
+inference_analysis = dm.analysis.InferenceLog(
+    timestamp_col=timestamp_col,
+    model_version_col=model_version_col,
+    prediction_col=prediction_col,
+    problem_type=problem_type,
+    label_col=label_col,
+    example_id_col=example_id_col
+)
+dm.create_or_update_monitor(
```

Comment:
I remember we discussed in the design doc but don't remember the conclusion: do we plan to give users a separate notebook for refreshing metrics, or do we want them to schedule and run this single one? If so, I think it'll be more clear if we run this with `skip_analysis=True` and then call `refresh_metrics` directly after. This helps the user understand the behavior better, compared to the implicit refresh we do on creation.

Reply:
We are depending on the instructions that users get from the private preview.